### ISSUE TITLE
Gekko is running strat.update and strat.check methods before strat.init method has completed

### ISSUE THREAD

**Issue by JoeTeglasi** [askmike/gekko#1807]
**Note: for support questions, please join our [Discord server](https://discord.gg/26wMygt)**

* **I'm submitting a ...**
[X] bug report
[?] feature request
[ ] question about the decisions made in the repository

* **Action taken** (what you did)
I'm trying to write a random forest algorithm that requires the generation of a lot of random numbers. for this I used the node-random npm package. 

* **Expected result** (what you hoped would happen)
I expected that Gekko would wait until all the code in the init method had finished running before beginning a backtest (before starting to feed candles through with the strat.update method)

* **Actual result** (unexpected outcome)
candles began to feed through immediately, by the time all of my random number matrices were generated, like 17,000+ candles had already passed through. Since my strategy hadn't finished initializing, those candles could not be used.

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc)

I don't know if this would be considered a bug, or more of a feature that would be nice to have, or simply a cry for help.

Please tell me how can I make the strategy wait to begin updating until the init function is complete?

**Comment by JoeTeglasi** [comment]
Issue has been resolved using “fibers/future” library:
(Please pardon my pseudo-code)

Global scope: 
Var Fiber = require(“fibers”);
Var Future = require(“fibers/future”);
var initComplete = false;
var functionChecklist={
Func1:”not-done”,
Func2:”not-done”,
Etc...
}
checkInitInt = setInterval(func(){
if (functionChecklist.Func1==“done” && functionChecklist.Func2==“done“ && etc..){
InitComplete.return(true);
}

strat.Init scope:
Func# callback = functionChecklist.Func# == “done“ 

strat.update scope:
Fiber(function(){
if(initComplete.wait()){

//do “update” stuff

}
}).run();

***do update stuff will wait to run until all Func#’s in strat.init are complete

**Comment by JoeTeglasi** [comment]
actually upon further inspection, it looks like the problem has been mitigated by using futures, but not completely resolved. instead of missing like 17k candles, it's now only missing around 2-3k.

the only reason the number of missed candles was lower was because adding the Fiber in the strat.update method meant that that method took more processing power and therefore slowed down the event loop, while the init method completed its functions asynchronously.

**Comment by JoeTeglasi** [comment]
issue was resolved by separately running a script to generate all the random numbers and fs.writeFileSync them into a .txt file, which then my strat.init function could read and assign to this.forest
